thumbor/loaders/file_fallback_http_loader.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# thumbor imaging service
+# https://github.com/thumbor/thumbor/wiki
+
+# Licensed under the MIT license:
+# http://www.opensource.org/licenses/mit-license
+# Copyright (c) 2011 globo.com timehome@corp.globo.com
+
+import re
+import tornado.httpclient
+from . import LoaderResult
+from thumbor.utils import logger
+from datetime import datetime
+from os import fstat
+from os.path import join, exists, abspath
+from urllib import unquote
+from tornado.concurrent import return_future
+from urlparse import urlparse
+from functools import partial
+
+@return_future
+def load(context, path, callback):
+    file_path = join(context.config.FILE_LOADER_ROOT_PATH.rstrip('/'), unquote(path).lstrip('/'))
+    file_path = abspath(file_path)
+    inside_root_path = file_path.startswith(context.config.FILE_LOADER_ROOT_PATH)
+
+    result = LoaderResult()
+
+    if inside_root_path and exists(file_path):
+
+        with open(file_path, 'r') as f:
+            stats = fstat(f.fileno())
+
+            result.successful = True
+            result.buffer = f.read()
+
+            result.metadata.update(
+                size=stats.st_size,
+                updated_at=datetime.utcfromtimestamp(stats.st_mtime)
+            )
+        callback(result)
+
+    elif path.startswith('http') == True:
+        load_sync(context, path, callback)
+
+    else:
+        result.error = LoaderResult.ERROR_NOT_FOUND
+        result.successful = False
+        callback(result)
+
+
+def validate(context, url):
+    if url.startswith('http') == True:
+        res = urlparse(url)
```

Comment:
This portion inside the `if` block should call the `http_loader.validate` to prevent code duplication